bpf/kmesh/ads/include/tail_call_index.h

```diff
@@ -0,0 +1,16 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
```

Comment:
As macro `KERNEL_VERSION_HIGHER_5_13_0` is removed from `kmesh_macros_def.h`, make `enum tail_call_index_t` as a standalone file prevents this compile issue:
```
In file included from ./bpf/kmesh/ads/include/kmesh_common.h:7,
                 from pkg/bpf/bpf_kmesh_common.go:23:
./bpf/include/bpf_common.h: In function 'record_manager_netns_cookie':
./bpf/include/bpf_log.h:113:18: error: 'KERNEL_VERSION_HIGHER_5_13_0' undeclared (first use in this function)
  113 |             if (!KERNEL_VERSION_HIGHER_5_13_0)                                                                         \
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```